api/server.js

```diff
@@ -1,3 +1,83 @@
 // BUILD YOUR SERVER HERE
+const express = require("express");
+const { find, findById, insert, update, remove } = require("./users/model.js");
 
-module.exports = {}; // EXPORT YOUR SERVER instead of {}
+const server = express();
+
+server.get("/api/users", async (req, res) => {
+  const users = await find();
+
+  if (users) {
+    return res.status(200).json({ users });
+  } else
+    return res
+      .status(500)
+      .json({ status: "The users information could not be retrieved" });
+});
+
+server.get("/api/users/:id", async (req, res) => {
+  const user = await findById(req.params.id);
+
+  if (user) {
+    return res.status(200).json({ user });
+  } else if (!user)
+    return res
+      .status(404)
+      .json({ status: "The user with the specified ID does not exist" });
+  else
```

Comment:
in acest bloc nu vei ajunge niciodata. Inteleg ce vrei sa faci. Deci uite 500, s-ar intampla daca la tine baza de date ar da fail de exemplu. avand functii asincrone, poti sa le dai lor wrap in trycatch, si in catch daca ceva nu merge vei returna 500. Exeplu:

```
try {
    const user = await findById(req.params.id);

    if (user) {
        return res.status(200).json({ user });
    } else  {
        return res
            .status(404)
            .json({ status: "The user with the specified ID does not exist" });
    }
} catch (error) {
    return res
      .status(500)
      .json({ status: "The users information could not be retrieved" });
}
```